### DISCUSSION TITLE
Executar função ao rolar a tela

### DISCUSSION THREAD

**Discussion by jhonathassouza** [frontendbr/forum#2416] [Perguntas e Respostas]
Boa tarde pessoal... estou precisando colocar uma música de fundo em uma página, porém o chrome bloqueia o autoplay. Eu consegui um código que da o play na música quando clica em um lugar específico que eu definir, porém gostaria que o play fosse executado ao rolar a página. Alguém pode me ajudar. Segue o código.

```
<script>
let audioButton = document.querySelector('.audio-button');   // <-- Change CSS Class Here
       audioButton.addEventListener('click', function(){
        var audio = document.getElementById("audio");

         if (audio.paused) {
            audio.play();
        }else{
            audio.pause();
            // audio.currentTime = 0
        }
      });
    </script>

    <audio id="audio" src=""></audio>
```

Onde tem o evento 'click' eu gostaria que fosse uma rolar de página, sei lá. Alguém tem alguma solução?

Obrigado

**Comment by viralatron** [comment]
já tive uma experiência breve com reprodução de áudio e eu decidi colocar o evento em um botão a fim de dar mais controle na reprodução ao usuário.

o motivo pelo qual os navegadores bloqueiam a reprodução automática é para evitar que o usuário seja pego desprevenido pelo som não solicitado.
se, mesmo assim, você busca colocar o som durante o rolar da página, você deveria a chamada de função em um evento de 'scroll'. sugiro que estude sobre eventos e como tratar eles para poder usar da melhor forma. 😄
